nova/virt/vmwareapi/vmops.py

```diff
@@ -2772,7 +2913,133 @@ def live_migration(self, instance, migrate_data, volume_mapping):
+                if not file_name:
+                    continue
+                if not file_name.endswith(CONFIGDRIVE_NAME):
+                    continue
+                dc_info = ds_util.get_dc_info(self._session, datastore)
+                ds_util.file_delete(self._session, file_name, dc_info.ref)
+            except vexc.VimException:
+                # No reason to error out the instance, as it is working
+                # but we need some visiblity here for the operator
+                LOG.exception("Cannot delete file", instance=instance)
+
+    def _disconnect_and_copy_cdrom(self, context, instance, cdrom, target,
+                                   dest_ds_ref):
+        backing = getattr(cdrom, "backing", None)
+        if not backing:
+            return None
+
+        file_name = getattr(backing, "fileName", None)
+        if not file_name:
+            return None
+
+        ds_ref = getattr(backing, "datastore", None)
+        if not ds_ref:
+            return None
+
+        get_path = ds_obj.DatastorePath.parse(file_name)
+
+        dest = target.prepare_ds_transfer(context, "PUT", dest_ds_ref,
+                                          get_path.rel_path)
+        ds_url = dest['ds_url']
+        ticket = dest['ticket']
+
+        if cdrom.connectable.connected:
+            self._force_disconnect_cdrom(instance, cdrom)
+
+        with contextlib.closing(self._get_ds_file_handle(
+                get_path, ds_ref, method='GET')) as get:
+            put = rw_handles.FileWriteHandle(ds_url,
+                    cookies=ticket,
+                    file_size=get.get_size())
+            with contextlib.closing(put):
+                shutil.copyfileobj(get, put)
+
+        ds_url = ds_obj.DatastoreURL.urlparse(ds_url)
+        ds_path = ds_obj.DatastorePath(ds_url.datastore_name, ds_url.path)
+        return ds_path
+
+    def _force_disconnect_cdrom(self, instance, cdrom):
+        vm_ref = vm_util.get_vm_ref(self._session, instance)
+        cf = self._session.vim.client.factory
+
+        backing = cf.create("ns0:VirtualCdromRemoteAtapiBackingInfo")
+        backing.deviceName = ""
+        backing.useAutoDetect = True
+
+        device_change = vm_util.create_virtual_cdrom_spec(cf, None,
+                cdrom.controllerKey, None,
+                cdrom.unitNumber, cdrom.key, backing)
+
+        config_spec = cf.create('ns0:VirtualMachineConfigSpec')
+        config_spec.deviceChange = [device_change]
+
+        reconfig_task = self._session._call_method(self._session.vim,
+                                                   "ReconfigVM_Task", vm_ref,
+                                                   spec=config_spec)
+        task_completed = threading.Event()
+
+        def set_task_completed(gt):
+            task_completed.set()
+
+        wait_for_task = utils.spawn(
+                self._session.wait_for_task, reconfig_task)
+        wait_for_task.link(set_task_completed)
```

Comment:
There is no check that the task succeded. Does the linked thread get called if the task failed? If no, we run into an endless loop. If yes, we will continue as if successfull.

Reply:
Yes, `set_task_completed` gets called on completion of the task, regardless of the status.
We will continue as successful, and the same error as before this patch will be raised.